src/Microsoft.AspNetCore.Mvc.Formatters.Json/JsonInputFormatter.cs

```diff
@@ -158,7 +158,18 @@ public class JsonInputFormatter : TextInputFormatter
 
                     if (successful)
                     {
-                        return InputFormatterResult.SuccessAsync(model);
+                        if (model == null && !context.AllowEmptyInput)
```

Comment:
Example cases, basically caused by pushing behaviour we didn't want in `BodyModelBinder` down into `JsonInputFormatter`:
1. Client submits an explicit JSON `null` body payload and a model binding feature described as disallowing "empty" input prevents binding.
2. Server property of type `int` is successfully bound to `0` but server property of type `string` is not when body contains whitespace. That sounds like the opposite of the expected behaviour.